### PR TITLE
Add minify option to defineFunction

### PR DESCRIPTION
**Pull request by fossamagna**
## Problem

**Issue number, if available:**

#2027

## Changes

1. Add `minify` option to `defineFunction`. Default value is `true`.

**Corresponding docs PR, if applicable:**

## Validation

Added test.

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/backend-function/API.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ export type FunctionProps = {
     environment?: Record<string, string | BackendSecret>;
     runtime?: NodeVersion;
     schedule?: FunctionSchedule | FunctionSchedule[];
+    minify?: boolean;
```
Comment (sobolk) [inline packages/backend-function/API.md:34]: Could you please create `FunctionBundlingOptions` and nest `minify` under it?

We'll be exposing more bundling options in the future so let's future proof it.

Comment (fossamagna) [reply packages/backend-function/API.md]: @sobolk
Thank you for your review and suggestion.
I have modified it as you suggested.

**Review by sobolk — DISMISSED** [review]

File: .changeset/pink-rockets-dance.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/backend-function': minor
+---
```
Comment (sobolk) [inline .changeset/pink-rockets-dance.md:3]:
```suggestion
---
'@aws-amplify/backend-function': minor
'@aws-amplify/backend': minor
---
```

**Review by sobolk — APPROVED** [review]

File: packages/backend-function/src/factory.ts [hunk 3]
```diff
@@ -298,6 +313,31 @@ class FunctionFactory implements ConstructFactory<AmplifyFunction> {
...
+      minify: this.resolveMinify(this.props.bundling),
+    };
+  };
+
+  private resolveMinify = (bundling?: FunctionBundlingOptions) => {
+    if (bundling === undefined) {
+      return true;
+    }
+
+    return bundling.minify === undefined ? true : bundling.minify;
```
Comment (Amplifiyer) [inline packages/backend-function/src/factory.ts:339]: can this be just?
```suggestion
    return bundling?.minify === undefined ? true : bundling.minify;
```

**Comment by sobolk** [comment]
Also.
I have ran the e2e test suite for this PR here https://github.com/aws-amplify/amplify-backend/actions/runs/11278596582/job/31367743355 .

There's one error that needs a fix:
image

Could you please include new function in this directory https://github.com/aws-amplify/amplify-backend/tree/main/packages/integration-tests/src/test-projects/data-storage-auth-with-triggers-ts/hotswap-update-files ?
These files are used to update test project, if function is not included there it gets deleted before assertions run.

**Review by sobolk — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**